### PR TITLE
plan: remove old validator logic

### PR DESCRIPTION
**Pull request by hanfei1991**
clean useless code. The validating has been done in rewriter function.
@shenli @coocood @zimulala @XuHuaiyu @tiancaiamao @winoros  PTAL

### CODE REVIEW

File: plan/expression_rewriter.go [hunk 1]
```diff
@@ -549,11 +550,26 @@ func (er *expressionRewriter) binaryOpToExpression(v *ast.BinaryOperationExpr) {
...
+		}
+		if er.err != nil {
+			return
+		}
 		function, er.err = expression.NewFunction(opcode.Ops[v.Op], &v.Type, er.ctxStack[stkLen-2:]...)
-	}
-	if er.err != nil {
-		er.err = errors.Trace(er.err)
-		return
+		if er.err != nil {
```
Comment (coocood) [inline plan/expression_rewriter.go]: Why move in switch?

File: plan/expression_rewriter.go [hunk 2]
```diff
@@ -538,6 +535,10 @@ func (er *expressionRewriter) unaryOpToExpression(v *ast.UnaryOperationExpr) {
 		er.err = errors.Errorf("Unknown Unary Op %T", v.Op)
 		return
 	}
+	if getRowLen(er.ctxStack[stkLen-1]) != 1 {
+		er.err = ErrOneColumn
+		return
+	}
```
Comment (hanfei1991) [inline plan/expression_rewriter.go]: because + (1,2) will be treated as  (1,2) instead of reporting error.

**Comment by coocood** [comment]
LGTM

File: plan/logical_plan_builder.go [hunk 3]
```diff
@@ -738,11 +738,17 @@ func (b *planBuilder) resolveGbyExprs(p LogicalPlan, gby *ast.GroupByClause, fie
...
 
 func (b *planBuilder) unfoldWildStar(p LogicalPlan, selectFields []*ast.SelectField) (resultList []*ast.SelectField) {
+	hasWildCard := false
 	for _, field := range selectFields {
 		if field.WildCard == nil {
 			resultList = append(resultList, field)
 			continue
 		}
+		if hasWildCard && field.WildCard.Table.L == "" {
+			b.err = ErrMultiWildCard
```
Comment (zimulala) [inline plan/logical_plan_builder.go]: This error message is ambiguous.

File: plan/expression_rewriter.go [hunk 4]
```diff
@@ -85,7 +85,7 @@ func constructBinaryOpFunction(l expression.Expression, r expression.Expression,
 	if lLen == 1 && rLen == 1 {
 		return expression.NewFunction(op, types.NewFieldType(mysql.TypeTiny), l, r)
 	} else if rLen != lLen {
-		return nil, errors.Errorf("Operand should contain %d column(s)", lLen)
+		return nil, ErrSameColumns
```
Comment (hanfei1991) [inline plan/expression_rewriter.go]: I will change it in next pr.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**